etc/di.xml

```diff
@@ -0,0 +1,88 @@
+            </argument>
+        </arguments>
+    </type>
+
+    <type name="Magento\Framework\EntityManager\Operation\ExtensionPool">
+        <arguments>
+            <argument name="extensionActions" xsi:type="array">
+                <item name="Magento\Catalog\Api\Data\ProductInterface" xsi:type="array">
+                    <item name="read" xsi:type="array">
+                        <item name="productlabelsReader" xsi:type="string">Smile\ProductLabel\Model\ProductLabel\ReadHandler</item>
+                    </item>
+                </item>
+            </argument>
+        </arguments>
+    </type>
+
+    <type name="Magento\Framework\EntityManager\MetadataPool">
+        <arguments>
+            <argument name="metadata" xsi:type="array">
+                <item name="Smile\ProductLabel\Api\Data\ProductLabelInterface" xsi:type="array">
+                    <item name="entityTableName" xsi:type="string">smile_productlabel</item>
+                    <item name="identifierField" xsi:type="string">product_label_id</item>
+                </item>
+            </argument>
+        </arguments>
+    </type>
+
+    <type name="Magento\Framework\EntityManager\HydratorPool">
+        <arguments>
+            <argument name="hydrators" xsi:type="array">
+                <item name="Smile\ProductLabel\Api\Data\ProductLabelInterface" xsi:type="string">Magento\Framework\EntityManager\AbstractModelHydrator</item>
+            </argument>
+        </arguments>
+    </type>
+
+    <type name="Magento\Framework\View\Element\UiComponent\DataProvider\CollectionFactory">
+        <arguments>
+            <argument name="collections" xsi:type="array">
+                <item name="smile_productlabel_productlabel_listing_data_source"
+                      xsi:type="string">Smile\ProductLabel\Model\ResourceModel\ProductLabel\Grid\Collection</item>
+            </argument>
+        </arguments>
+    </type>
+
+    <type name="Smile\ProductLabel\Model\ResourceModel\ProductLabel\Grid\Collection">
+        <arguments>
+            <argument name="mainTable" xsi:type="string">smile_productlabel</argument>
+            <argument name="eventPrefix" xsi:type="string">smile_productlabel_grid_collection</argument>
+            <argument name="eventObject" xsi:type="string">smile_productlabel_grid_collection</argument>
+            <argument name="resourceModel" xsi:type="string">Smile\ProductLabel\Model\ResourceModel\ProductLabel</argument>
+        </arguments>
+    </type>
+
+    <type name="Smile\ProductLabel\Controller\Adminhtml\ImageLabel\Image">
+        <arguments>
+            <argument name="imageUploader" xsi:type="object">Magento\Catalog\CategoryImageUpload</argument>
+        </arguments>
+    </type>
+    <virtualType name="Magento\Catalog\CategoryImageUpload" type="Magento\Catalog\Model\ImageUploader">
```

Comment:
Warning, the virtualType with the same name is existing on the `Magento\Catalog` module. You cannot re-declare it here, you should declare a virtualType which is named like `Smile\ProductLabel\ProductLabelImageUpload` and inject it on line 72.

Reply:
Done.